datafusion/expr/src/built_in_function.rs

```diff
@@ -884,8 +885,13 @@ impl BuiltinScalarFunction {
                 ],
                 self.volatility(),
             ),
-            BuiltinScalarFunction::DateTrunc => Signature::exact(
-                vec![Utf8, Timestamp(Nanosecond, None)],
+            BuiltinScalarFunction::DateTrunc => Signature::one_of(
```

Comment:
this clause seems unreachable given the change above to match `BuiltinScalarFunction::DateTrunc | BuiltinScalarFunction::DateBin => {`

I would personally suggest keeping this change and removing the `BuiltinScalarFunction::DateTrunc | BuiltinScalarFunction::DateBin => {` change as it make the code more explicy